src/gluonts/nursery/streaming_models/forecast_generator.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import logging
+from typing import Any, Iterator, List, Optional
+
+import mxnet as mx
+import numpy as np
+
+from gluonts.core.component import validated
+from gluonts.dataset.field_names import FieldName
+from gluonts.dataset.loader import DataLoader, InferenceDataLoader
+from gluonts.model.forecast_generator import ForecastGenerator, OutputTransform
+from gluonts.mx.distribution import DistributionOutput
+from gluonts.mx.model.forecast import DistributionForecast
+
+from .predictor import NETWORK_STATE_KEY, PREDICTOR_STATE_KEY
+
+
+def to_numpy(x: Any) -> Any:
+    if isinstance(x, mx.nd.NDArray):
+        return x.asnumpy()
+    if isinstance(x, np.ndarray):
+        return x
+    if isinstance(x, (list, tuple)):
+        if isinstance(x[0], (float, int, np.float)):
+            return x
+        else:
+            tmp = [to_numpy(xi) for xi in x]
+            if isinstance(x, tuple):
+                return tuple(tmp)
+            return tmp
+    return x
+
+
+class StatefulDistributionForecastGenerator(ForecastGenerator):
+    @validated()
```

Comment:
I wonder, can we get around not using `validated`? Does `core.ty.Stateless` work?